newIDE/app/src/EventsSheet/ParameterFields/GenericExpressionField/index.js

```diff
@@ -95,7 +95,10 @@ export default class ExpressionField extends React.Component<Props, State> {
   }
 
   focus() {
-    if (this._field) this._field.focus();
+    if (this._field) {
+      this._field.focus();
+      this._doValidation();
```

Comment:
Can you triple check if there is no weird side effect? Validation can also take a bit of time during which it can froze the UI, so running it on focus might be risky? I would rather use `this._enqueueValidation`. `this._enqueueValidation` will trigger a validation 500ms after the last input. This means that the user can start typing immediately - without having a lag in case the validation is taking too much time - or can just wait half a second and see any errors showing.

(of course if the validation was instant it would not be a problem, but we must be prepared for the case where the validation is taking a few milliseconds, which can be noticeable to the user eye, especially if we're blocking typing)

Reply:
updated now :)